### DISCUSSION TITLE
Create React Native App

### DISCUSSION THREAD

**Discussion by linktel0** [matter-js/matter.js#828] [Feature requests (vote via thumbs up)]
Can you create a repository about example of the React Native app?
npx  react-native init myapp --template react-native-template-typescript
npm i @Project-chip/Matter-node.js.

**Comment by Apollon77** [comment]
I like that idea. As described in the readme at https://github.com/project-chip/matter.js?tab=readme-ov-file#how-to-use-matterjs-in-own-projects there are some "platform specific parts" that needs to be implemented. These are implemented for node.js with the matter-node.js package, but this is not "the same" for react-native.

This means that someone with knowledge in this area should create a package like "matter-react-native.js" that implements the relevant parts like network, crypto, timer, logging and such using react-native. Then this can be used. The matter-node.js package is a good blueprint on what needs to be implemented.

I myself experimented around a bit with implementing the bluetooth part with a react-native lib, but thats it. I might add this somewhen, but I also can not test this myself honestly.

I would also be very open and happy about someone contributing this as a PR as a new package for the matter.js monorepo, but I personally do not plan to implement this in the next time because we have many things still left open on the Matter protocol level and compliance. So this topic is ideas as a contribution by someone which is more familiar with react-native then I :-)

Ingo
